src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1058,10 +1058,18 @@
                               (map #(vector certname-id %) new-package-ids)))
 
         (when (seq old-package-ids)
-          (jdbc/delete! :certname_packages
-                        ["certname_id = ? and package_id = ANY(?)"
-                         certname-id
-                         (sutils/array-to-param "bigint" Long (map long old-package-ids))]))))))
+          (let [old-package-id-sql-param (sutils/array-to-param "bigint" Long
+                                                                (map long old-package-ids))]
```

Comment:
You tell me, it's your code :stuck_out_tongue: 

I actually looked into this at wyatt's request for the first round, but I don't think I reached any conclusion. Ended up just leaving it as-is since it worked.